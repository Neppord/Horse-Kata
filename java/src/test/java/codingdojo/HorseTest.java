package codingdojo;


import org.approvaltests.Approvals;
import org.junit.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Optional;

import static org.junit.Assert.assertEquals;

public class HorseTest
    {
        @Test
        public void FilterSortPaginate_No_Filters_No_Sorting_No_Pagination()
        {
            // Arrange - this is data from another service or database
            List<String> headers = SampleHorseData.GetSampleHeaders();
            List<List<Object>> tableData = SampleHorseData.GetSampleTableData();

            // These objects describe the query we got from the front end
            List<FilterMetadata> filters = Collections.emptyList();
            Optional<SortMetadata> sortMetadata = Optional.empty();
            PaginationMetadata paginationMetadata = new PaginationMetadata(0, 10);

            // Act
            PaginatedTable table = Horse.FilterSortPaginateTable(headers, tableData, filters, sortMetadata, paginationMetadata);

            // Assert the data to be sent to the front end
            assertEquals(Arrays.asList("Breed", "Colour", "Height", "Age", "Shoes"), table.headers);
            assertEquals(5, table.totalRows);
            Approvals.verifyAll("tableData", table.tableData);
        }
    }

