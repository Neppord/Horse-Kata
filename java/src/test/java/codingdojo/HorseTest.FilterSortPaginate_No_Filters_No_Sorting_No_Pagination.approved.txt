tableData[0] = [Thoroughbred, Bay, 1.6, 3, true]
tableData[1] = [Thoroughbred, Grey, 1.55, 3, true]
tableData[2] = [Arabian horse, Bay, 1.51, 5, true]
tableData[3] = [Shetland Pony, Black, 1.01, 2, false]
tableData[4] = [Shire horse, Black, 1.71, 4, true]
